crates/fluvio-sc/src/controllers/mirroring/controller.rs

```diff
@@ -0,0 +1,220 @@
+                                            .partitions()
+                                            .iter()
+                                            .position(|rc| rc.remote_cluster == mirror.remote_id);
+
+                                        if partition.is_none() {
+                                            return Err(anyhow::anyhow!(
+                                                "Topic {} is not a mirror home for cluster {}",
+                                                topic.key,
+                                                mirror.id
+                                            ));
+                                        }
+
+                                        let partition = partition.unwrap() as u32;
+
+                                        let topic_replica_map = &topic.replica_map;
+                                        let home_spu_id = topic_replica_map
+                                            .get(&partition)
+                                            .context(
+                                                "Topic does not have a replica for {partition}",
+                                            )?
+                                            .first()
+                                            .context("Topic does not have any replicas")?;
+
+                                        let replica: ReplicaSpec = ReplicaSpec::Mirror(
+                                            MirrorConfig::Remote(RemoteMirrorConfig {
+                                                home_spus: vec![*home_spu_id; 1],
+                                                home_cluster: mirror.id.clone(),
+                                            }),
+                                        );
+
+                                        let mut remote_topic: TopicSpec = replica.into();
+                                        if let Some(cleanup_policy) = topic.spec.get_clean_policy()
+                                        {
+                                            remote_topic.set_cleanup_policy(cleanup_policy.clone())
+                                        }
+
+                                        remote_topic.set_compression_type(
+                                            topic.spec.get_compression_type().clone(),
+                                        );
+
+                                        remote_topic.set_deduplication(
+                                            topic.spec.get_deduplication().cloned(),
+                                        );
+
+                                        if let Some(storage) = topic.spec.get_storage() {
+                                            remote_topic.set_storage(storage.clone());
+                                        }
+                                        remote_topic
+                                    } else {
+                                        return Err(anyhow::anyhow!("Topic is not a mirror home"));
```

Comment:
This need integration test

Reply:
Integration test or E2E?

Reply:
integration test is nice to have but E2E is must

Reply:
Ok, I will add E2E test in this issue: https://github.com/infinyon/fluvio/issues/3972